### PR TITLE
fix: don't modify contract principals in postconditions

### LINKED ISSUE #1457: NFT post-condition for contract overwritten
**Describe the bug**

Transaction failed due to post-conditions even though correct post-conditions were provided in `doContractCall`.

Transaction ID: 0xf4af5cec139fac67617fbd67a1566afac698b825e237aa6a555b0c7628191f8f on testnet

**To Reproduce**
1. Call a contract with 
```
makeContractNonFungiblePostCondition(
            CONTRACT_ADDRESS,
            CONTRACT_NAME,
            NonFungibleConditionCode.DoesNotOwn,
            createAssetInfo(nftContractAddress, nftContractName, nftAssetName),
            nftIdCV
          ),
```
2. See failure of tx due to post conditions and post condition set for tx-sender, not (escrow) contract.

**Expected behavior**
Post-conditions should be applied unchanged.

**Screenshots**
Screenshot from 2021-07-20 15-32-23

### PR DESCRIPTION
**Pull request by aulneau**
> Try out this version of the Stacks Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/1052861729).

resolves https://github.com/blockstack/stacks-wallet-web/issues/1457

@friedger can you test?

### CODE REVIEW

**Comment by friedger** [comment]
The wording of the post-condition seems incorrect. The receiver of the NFT is not known but shown as the contract deployer.

Screenshot from 2021-07-20 17-04-01

**Comment by aulneau** [comment]
> The wording of the post-condition seems incorrect. The receiver of the NFT is not known but shown as the contract deployer.
> 
> Screenshot from 2021-07-20 17-04-01

Thanks for this-- just pushed a new fix, can you test 1 more time?

**Comment by friedger** [comment]
Now the word `from` is missing
Screenshot from 2021-07-20 18-01-26

**Comment by aulneau** [comment]
> Now the word `from` is missing
> Screenshot from 2021-07-20 18-01-26

yeah i decided to remove the `from` or `to` as the to is never really knowable

File: src/common/transactions/postcondition-utils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    const transformedPostCondition = handlePostConditions(
+      [postCondition],
+      SENDER_ADDRESS,
+      SENDER_ADDRESS
+    );
+
+    expect(transformedPostCondition[0]).toEqual(postCondition);
+  });
+
+  it('should not modify a post condition addresses are all the same', () => {
```
Comment (fbwoolf) [inline src/common/transactions/postcondition-utils.spec.ts]: Not sure this statement reads correctly?

Comment (aulneau) [reply src/common/transactions/postcondition-utils.spec.ts]: yeah should be something like
```suggestion
  it('should not modify a post condition when all addresses are the same (sender, current address, and postcondition principal)', () => {
```

**Review by beguene — APPROVED** [review]

**Review by fbwoolf — APPROVED** [review]

**Pull request merged**